portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,7 +27,29 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Tino!</h1>");
+    ArrayList<String> comments = new ArrayList<String>();
+    comments.add("Hi tino");
+    comments.add("i like pie");
+    comments.add("nice space");
+
+    String json = convertToJson(comments);
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
   }
+
+    /**
+      * Converts a data array instance into a JSON string using manual String concatentation.
+    */
+  private String convertToJson(ArrayList<String> array) {
+    String json = "";
+    json += "\"" + array.get(0) + "\"";
+    json += ", ";
+    json += "\"" + array.get(1) + "\"";
+    json += ", ";
+    json += array.get(2);
+    return json;
```

Comment:
use String format instead.

Reply:
Also this assumes the array has 3 elements. Use for loop will be another option

Reply:
Please use Gson library for this: https://stackoverflow.com/questions/14228912/how-to-convert-list-to-json-in-java.